lib/netext/httpext/response.go

```diff
@@ -158,11 +169,38 @@ func (res *Response) JSON(selector ...string) (interface{}, error) {
 		}
 
 		if err := json.Unmarshal(body, &v); err != nil {
+			switch t := err.(type) {
+			case *json.SyntaxError:
+				err = checkErrorInJSON(body, int(t.Offset), err)
+			default:
+				break
+			}
 			return nil, err
 		}
 		res.validatedJSON = true
 		res.cachedJSON = v
 	}
 	return res.cachedJSON, nil
+}
+
+func checkErrorInJSON(input []byte, offset int, err error) error {
+	lf := '\n'
+	str := string(input)
+
+	// Humans tend to count from 1.
+	line := 1
+	character := 0
+
+	for i, b := range str {
+		if b == lf {
```

Comment:
Currently there is no test for this, so I would be glad if you can make one but you will probably need to make a custom handler like in https://github.com/loadimpact/k6/blob/3fc300cd779d9ce775eb299d0f0dbde7a620ce75/js/modules/k6/http/request_test.go#L1363-L1384 and just return a broken json on line 5 for example :)